Jessica Minnier

Package Review
--------------

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

-   \[X\] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor *before* starting your review).

#### Documentation

The package includes all the following forms of documentation:

-   \[X\] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README
-   \[X\] **Installation instructions:** for the development version of package and any non-standard dependencies in README
-   \[X\] **Vignette(s)** demonstrating major functionality that runs successfully locally
-   \[X\] **Function Documentation:** for all exported functions in R help
-   \[X\] **Examples** for all exported functions in R Help that run successfully locally
-   \[ \] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

#### Functionality

-   \[X\] **Installation:** Installation succeeds as documented.
-   \[X\] **Functionality:** Any functional claims of the software been confirmed.
-   \[X\] **Performance:** Any performance claims of the software been confirmed.
-   \[X\] **Automated tests:** Unit tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
-   \[ \] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines

#### Final approval (post-review)

-   \[ \] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 4 (Including time spent reading example reviews, I'm new to this!)

------------------------------------------------------------------------

### Review Comments

This package provides an interface to download sequencing data from the 1KP project. There are just a few functions, mainly to retreive the full data set, filter by various types of data queries, and then download the sequencing data based on the full data or on a filter. Overall the package works as intended and it is very useful to the community. I would recommend the package be accepted. My comments are all minor.

**Documentation:**

-   I like the print properties of the `OneKP` object! There is no example in the `print.OneKP` function help page, but I don't think that is necessary? I'm not sure if that is typical.
-   In the vignette and in the example for `filter_by*`, it would be helpful to show how to list the codes/species that you can filter on. I realize it is accessed in `onekp@table$code` but it took me a second to realize that.
-   In the `?download` documentation, you say "These functions will return all files in the OneKP object of the given type (e.g. protein FASTA files for get\_peptides)." Since there are only two options, peptides and nucleotides, can you say the file type for `get_nucleotides`? (i.e. what is an .fna file?)
-   The `?oneKP` description is short, would be nice to have 1KP link here. \[overlaps with @zachary-foster review\]
-   There isn't a separate vignette but the readme serves as one.

**Regarding packaging guidelines:**

-   strongly recommend package name in lower case \[overlaps with @zachary-foster review\]
-   Nice description in the `README`.
-   I don't see a `NEWS` file? \[overlaps with @zachary-foster review\]
-   98% of code is covered by tests
-   Contributing guidelines are mostly fulfilled but it would be nice to have a `contributing.md` file to explain how to contribute to the package. \[overlaps with @zachary-foster review\]

**Functions:**

-   For the `download` functions, it would be useful to include an option set the location of files. It also might be helpful to have an option to return the full path to files as opposed to relative paths.

This code returns an error, which is good, but it would be useful to return a more clear error. Filtering on a non-existant species results in a 0 "length" oneKP object (ok), but then the `download_*` functions return a curl error due to a misspecified url.

    tmp <- filter_by_species(onekp,"foobar") # returns object with 0 0
    tmp
    str(tmp@table)

    download_nucleotides(tmp)
    > Error in curl::curl_download(url = url, destfile = path, quiet = TRUE) : Argument 'url' must be string.
